### PR TITLE
check markers

### PR DESCRIPTION
**Pull request by AurelienRenou**
check if there is nan values in the c3d and delete lines which correspond in the difference vector and the jacobian matrix

### CODE REVIEW

File: static_inverse_kinematics.py [hunk 1]
```diff
@@ -41,6 +40,10 @@ class StaticInverseKinematics:
 
     Methods
     -------
+    get_marker_trajectories(self)
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: private or not ?

File: static_inverse_kinematics.py [hunk 2]
```diff
@@ -63,21 +66,46 @@ def __init__(
...
 
+        self._get_idx_to_remove()
+
         self.q = np.zeros((self.nb_q, self.nb_frames))
-        self.bounds_min, self.bounds_max = np.squeeze(get_range_q(self.biorbd_model))
         self.bounds = get_range_q(self.biorbd_model)
-        
-    def _marker_diff(self, q: np.ndarray, xp_markers: np.ndarray):
+
+    def get_marker_trajectories(self) -> np.ndarray:
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: I think it should be private

File: static_inverse_kinematics.py [hunk 3]
```diff
@@ -63,21 +66,46 @@ def __init__(
         self.c3d_path_file = c3d_path_file
```
Comment (Ipuch) [inline static_inverse_kinematics.py:68]: Is it used elsewhere? If not delete the line.

File: static_inverse_kinematics.py [hunk 4]
```diff
@@ -63,21 +66,46 @@ def __init__(
...
             self.biorbd_model.markerNames()[i].to_string() for i in range(len(self.biorbd_model.markerNames()))
         ]
-        self.xp_markers = self.c3d_data.trajectories
+        self.xp_markers = self.get_marker_trajectories()
         self.nb_q = self.biorbd_model.nbQ()
-        self.nb_frames = self.c3d_data.nb_frames
+        self.nb_frames = self.c3d["parameters"]["POINT"]["FRAMES"]["value"][0]
         self.nb_markers = self.biorbd_model.nbMarkers()
 
+        self._get_idx_to_remove()
```
Comment (Ipuch) [inline static_inverse_kinematics.py:81]: idx_to_remove is missing in the docstring

File: utils.py [hunk 5]
```diff
@@ -65,3 +66,27 @@ def get_segment_and_dof_id_from_global_dof(biorbd_model: biorbd.Model, global_do
...
+    """
+    factor_str = c3d_file["parameters"]["POINT"]["UNITS"]["value"][0]
+    if factor_str == "mm":
+        factor = 1000
+    elif factor_str == "m":
+        factor = 1
+    else:
+        raise NotImplementedError("This is not implemented for this unit")
+
+    return factor
```
Comment (Ipuch) [inline utils.py]: code is not blacked

**Review by Ipuch — COMMENTED** [review]
Very tiny details this is much clearer! Great work ;) Don't forget to black and I'll accept the PR :)

**Comment by Ipuch** [comment]
@AurelienRenou please when you do a pull request very if it is related to an issue and mention it in the PR:
related to #5

File: static_inverse_kinematics.py (hunk 2 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 3 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py:68]: it is used to load the c3d

File: static_inverse_kinematics.py (hunk 4 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py:81]: done

File: utils.py (hunk 5 above)
Comment (AurelienRenou) [reply utils.py]: done

**Pull request merged**